client/signup/navigation-link/index.jsx

```diff
@@ -53,10 +53,17 @@ export class NavigationLink extends Component {
 	 */
 	getPreviousStep() {
 		const { flowName, signupProgress, stepName } = this.props;
+		const steps = getFilteredSteps( flowName, signupProgress );
 
-		let steps = getFilteredSteps( flowName, signupProgress );
-		steps = steps.slice( 0, findIndex( steps, step => step.stepName === stepName ) );
-		const previousStep = findLast( steps, step => ! step.wasSkipped && step.stepName !== stepName );
+		if ( ! isArray( steps ) ) {
+			return { stepName: null };
+		}
+
+		const previousSteps = steps.slice( 0, findIndex( steps, step => step.stepName === stepName ) );
+		const previousStep = findLast(
+			previousSteps,
+			step => ! step.wasSkipped && step.stepName !== stepName
```

Comment:
As @p-jackson mentioned, we could remove `step.stepName !== stepName` as that's accounted for in the `slice` in the previous line

Reply:
Taken care of in https://github.com/Automattic/wp-calypso/pull/35049/commits/b569dab205e97c98d816ee251ab1c5b9bd697620